subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/execution/PluginsBlockInterpreter.kt

```diff
@@ -56,68 +54,135 @@ fun interpret(program: Program.Plugins): PluginsBlockInterpretation {
 }
 
 
+@Suppress("unused")
+private
+val debugger: ParserDebugger = ParserDebugger()
+
+
+@Suppress("UNUSED_PARAMETER")
+private
+fun <T> debug(name: String, parser: Parser<T>) =
```

Comment:
:x: The `ParserDebugger` is duplicated in `KotlinGrammar`, this should be fixed. We could decide to simply remove this in order not to clutter the production code but I understand how keeping it around is interesting for future work. 

🎨 I'm also concerned by the fact that the usage is not very DRY as you have to repeat a variable name in a string on each usage. This can be solved by using Kotlin property delegates. Let's talk about this.

Reply:
Fixed it.